src/main/java/winstone/Launcher.java

```diff
@@ -229,8 +224,11 @@ private void writePortToFileIfNeeded() throws IOException {
                     int port = ((ServerConnector) connector).getLocalPort();
                     Path portFile = Paths.get(portFileName);
                     Path portDir = portFile.getParent();
+                    if (portDir == null) {
+                        throw new IllegalArgumentException("Given port file name doesn't have a parent: " + portFileName);
+                    }
                     Files.createDirectories(portDir);
-                    Path tmpPath = Files.createTempFile(portDir, portFile.getFileName().toString(), null);
+                    Path tmpPath = Files.createTempFile(portDir, portFileName, null);
```

Comment:
this cause `java.lang.IllegalArgumentException: Invalid prefix or suffix` as now the full path (if the user provide a full absolute path such as the unit test) is used as prefix which is not not possible.
ignore this fake/false positive spotbugs as there are many checks on this just few lines before.